session/manager_http.go

```diff
@@ -273,9 +273,8 @@ func (s *ManagerHTTP) DoesSessionSatisfy(r *http.Request, sess *Session, request
 			if err != nil {
 				return err
 			}
-			sess.Identity = i
 		} else if len(i.Credentials) == 0 {
-			// If credentials are not expanded, we load them here.
+			i = sess.Identity.CopyWithoutCredentials() // we don't want to modify the identity in the session object
 			if err := s.r.PrivilegedIdentityPool().HydrateIdentityAssociations(ctx, i, identity.ExpandCredentials); err != nil {
 				return err
 			}
```

Comment:
The default identity JSON marshaller excludes credentials and metadata from the JSON response. This method was a legacy approach where we did not have that functionality yet.

My recommendation is to hydrate the session identity, because then we avoid calling the same code again later (in case we need the credentials somewhere else too).

Reply:
Got it, changed.